physionet-django/project/modelcomponents/publishedproject.py

```diff
@@ -203,22 +203,24 @@ def has_access(self, user):
         if not self.allow_file_downloads:
             return False
 
-        if self.access_policy == AccessPolicy.CREDENTIALED and (
-            not user.is_authenticated or not user.is_credentialed
-        ):
-            return False
-        elif self.access_policy == AccessPolicy.RESTRICTED and not user.is_authenticated:
-            return False
-
-        if self.is_self_managed_access:
-            return DataAccessRequest.objects.filter(
-                project=self, requester=user,
-                status=DataAccessRequest.ACCEPT_REQUEST_VALUE).exists()
-        elif self.access_policy:
-            return DUASignature.objects.filter(
-                project=self, user=user).exists()
+        if self.access_policy == AccessPolicy.OPEN:
+            return True
+        elif self.access_policy == AccessPolicy.RESTRICTED:
+            return user.is_authenticated and DUASignature.objects.filter(project=self, user=user).exists()
+        elif self.access_policy == AccessPolicy.CREDENTIALED:
+            return (
+                user.is_authenticated
+                and user.is_credentialed
+                and DUASignature.objects.filter(project=self, user=user).exists()
+            )
+        elif self.access_policy == AccessPolicy.CONTRIBUTOR_REVIEW:
```

Comment:
There seems to be an issue with this `elif` chunk. If a user's initial request for access is rejected, it seems they can never gain access. To reproduce:

1. As "Admin", create a contributor reviewed project.
2. As a new user, request access to the data
3. As Admin, reject the request.
4. As the new user, request access to the data.
5. As Admin, approve the request.
6. The new user is unable to access the data.

Reply:
The issue was in the template. Probably the new user isn't credentialed.
![image](https://user-images.githubusercontent.com/44640544/152967750-e603b913-50dc-4334-8f4a-dc2040a0bb68.png)